src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2136,12 +2135,13 @@ private static IEnumerable<Diagnostic> FilterDiagnosticsByLocation(IEnumerable<D
             }
         }
 
-        internal ImmutableArray<Diagnostic> GetDiagnosticsForSyntaxTree(
+        internal override ImmutableArray<Diagnostic> GetDiagnosticsForSyntaxTree(
             CompilationStage stage,
             SyntaxTree syntaxTree,
             TextSpan? filterSpanWithinTree,
             bool includeEarlierStages,
-            CancellationToken cancellationToken = default(CancellationToken))
+            bool includeDiagnosticsWithSourceSuppression,
```

Comment:
Reasoning being this flag is just a filter applied at end of reporting diagnostics. So, anyone wanting suppressed diagnostics is likely interested in getting it for all APIs. If not, then can filter them out for APIs they don't care about.

Primary reason for making it a compilation option is that this way even command line builds can emit source suppressed compiler diagnostics in error log, and eventually if we add a new command line switch, dump it to console output.